editor/components/post-publish-panel/index.js

```diff
@@ -56,25 +57,32 @@ class PostPublishPanel extends Component {
 		}
 	}
 
-	onPublish() {
+	onSubmit() {
+		const { user, onClose } = this.props;
+		const userCanPublishPosts = get( user.data, [ 'post_type_capabilities', 'publish_posts' ], false );
+		const isContributor = user.data && ! userCanPublishPosts;
+		if ( isContributor ) {
```

Comment:
Nicely moved to one place. I hope we move that logic to selector one day. Not possible  at the moment without a bigger refactor :)